streams/src/main/java/org/apache/kafka/streams/processor/internals/GlobalStateManagerImpl.java

```diff
@@ -99,7 +100,7 @@ public void setGlobalProcessorContext(final InternalProcessorContext processorCo
     @Override
     public Set<String> initialize() {
         try {
-            if (!stateDirectory.lockGlobalState()) {
+            if (stateDirectory.isGlobalStateDirExists() && !stateDirectory.lockGlobalState()) {
```

Comment:
Similar here: we could fix this within `lockGlobalState()` and just return `true` if the state directory was not created.

Reply:
ack.